src/clean_air/data/data_subset.py

```diff
@@ -114,16 +113,33 @@ def extract_box(self, box, crs=None):
 
         return cube
 
-    def extract_track(self, track, crs=None):
+    def extract_track(self, start=None, end=None):
         """
         Extract a track
 
         Arguments:
-            track: timeseries of the track, as a dataframe
+            start (datetime.time or str): initial time filter limit
+            end (datetime.time or str): end time filter limit
         """
         cube = self._load_cube()
 
-        return util.cubes.extract_series(cube, track)
+        if isinstance(self.metadata, str):
+            ds = xr.open_dataset(self.metadata)
+        elif isinstance(self.metadata, iris.cube.Cube):
+            ds = xr.DataArray.from_iris(self.metadata)
+
+        df = ds.to_dataframe()
+        if start and end:
+            df = df.between_time(start, end)
+            if df.empty:
+                raise ValueError('Empty dataframe, likely due to time bounds being out of range')
+
+        series = util.cubes.extract_series(cube, df, column_mapping={'time': 'time'})
+        output_cube = iris.pandas.as_cube(series)
+        output_cube.rename(cube.name())
+        output_cube.units = cube.units
+
+        return output_cube
```

Comment:
There's a lot of flipping around from one type of dataset to another here, could you just explain what's going on here please?

Reply:
Passed to `extract_series` is the original iris cube, and a pandas dataframe version of the data that is cut down to the time period we're interested in. Within `extract_series`, these are used to interpolate the cube along the track, that is then converted and returned as a pandas series. `extract_track` then converts this back to a cube. 

So for this particular case the final switch to a series and back again isn't needed - but it's using a more general helper function that is designed to extract a series from a cube, so I think a good idea to leave that as returning a series.

On the other hand! It does take some time to run; ~1 min for data with 1000 timesteps. So maybe it's a good idea to rethink this process from the ground up, and ignore what was written previously? Slightly difficult to say without knowing other potential use cases for the functions involved.

(Sorry for the stream of consciousness rambling!)

Reply:
I'm still wondering what the use case would be in which we need this to be converted to a pandas series and then back again.  Let's chat about it later.